oat-sa/extension-tao-review
# Feature/tao 8492/refactor test review

### Summary
Related to: https://oat-sa.atlassian.net/browse/TAO-8492

Refine the test review, fixing the navigation and integrating the review panel.
Please note that the UI design flaws will be fixed in another PR. This one only addresses the technical debt.

### How to test in situation
The extension only works through LTI. To test the feature in situation, you'll need to:
- install the extension `taoLti`, along with this one
- setup a LTI provider
- compile a delivery and take the assessment through LTI
- finally, access the review through LTI

In order to call the LTI link, you may use this useful simulator: http://ltiapps.net/test/tc.php

To get the LTI link to the delivery, from the Deliveries backoffice in TAO, select your delivery and click on the LTI button, then copy the link:
<img width="800" alt="image" src="https://user-images.githubusercontent.com/1500098/62601264-3bb72c00-b8f1-11e9-9347-a353d6e3155b.png">

On the simulator page, fill in the following fields:
- Launch URL:  the link copied from the delivery
- Consumer key: the key setup in the LTI provider
- Shared secret: the secret setup in the LTI provider
<img width="691" alt="image" src="https://user-images.githubusercontent.com/1500098/62601396-8f297a00-b8f1-11e9-8bb5-8e8e03b4a777.png">
- Roles: should be `Learner`
<img width="502" alt="image" src="https://user-images.githubusercontent.com/1500098/62601449-ab2d1b80-b8f1-11e9-9a81-93ada82a1211.png">

Click on the button `Save data`, then on `Launch TP in new window`, and pass the assessment.
On the LTI return page, copy the URI of the the test execution:
<img width="934" alt="image" src="https://user-images.githubusercontent.com/1500098/62601715-50e08a80-b8f2-11e9-8b3a-a7deb153a02f.png">

Go back to the LTI emulator, and replace the Launch URL by `http://<YOUR_HOST>/taoReview/ReviewTool/launch?execution=<URI>` (replace `<YOUR_HOST>` by you local domain name, and `<URI>` by the URI copied from the return page).

Click on the button `Save data`, then on `Launch TP in new window`, and you should see the test review in action.

### Unit testing
Impacted unit tests:
- `taoReview/views/js/test/review/component/qtiTestReviewComponent/test.html`
- `taoReview/views/js/test/review/config/qtiTestReviewConfig/test.html`
- `taoReview/views/js/test/review/plugins/navigation/next-prev-review/test.html`
- `taoReview/views/js/test/review/plugins/navigation/review-panel/plugin/test.html`
- `taoReview/views/js/test/review/provider/qtiTestReviewProvider/test.html`
- `taoReview/views/js/test/review/proxy/qtiTestReviewProxy/test.html`

Run the unit tests:
```
cd tao/views/build
npx grunt connect:dev taoreviewtest
```
